client/src/main/java/io/pravega/client/tables/KeyUpdateListener.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.client.tables;
+
+import java.util.function.Consumer;
+
+/**
+ * Defines a Listener for Key Updates.
+ *
+ * @param <KeyT>   Key Type.
+ * @param <ValueT> Value Type.
+ */
+public interface KeyUpdateListener<KeyT, ValueT> extends AutoCloseable {
+    /**
+     * Gets a value indicating the Key that this Listener is attached to.
+     * @return The Key.
+     */
+    KeyT getListeningKey();
+
+    /**
+     * This will be invoked whenever an Entry matching {@link #getListeningKey()} is inserted or updated.
+     * @param value A Consumer that, when invoked, will process the given value update.
+     */
+    void handleValueUpdate(Consumer<GetResult<ValueT>> value);
```

Comment:
@tkaitchuck If it were future based, then the caller would have to invoke a method which returns a Future which we'd complete. It's possible that we can miss updates using this method.

The notification approach registers something on the server, which will then send the message back to the client. 

I agree with the blocking problem. I'll add an `Executor` arg to these methods.

Reply:
@shiveshr 
We can choose what the approach is. We can either queue all updates and invoke this for each (in order), or just invoke it for the last update.

I can add one for the whole Table too. I was reluctant to doing so because for hot tables, that will mean a lot of chatter.

Reply:
I have rewritten this API to:
1. Allow either a set of Keys or the entire Table/TableSegment.
2. Accept an Executor on which to invoke callbacks.
3. Get notified when the table is sealed/modified/deleted
4. Wire the `close` method to an "unregister" (subject to implementation; not visible in API)
5. Updated Javadoc to explain what happens with all updates.